thorlcr/activities/lookupjoin/thlookupjoinslave.cpp

```diff
@@ -2264,7 +2286,7 @@ class CLookupJoinActivityBase : public CInMemJoinBase<HTHELPER, IHThorHashJoinAr
             if (getOptBool(THOROPT_LKJOIN_HASHJOINFAILOVER)) // for testing only (force to disk, as if spilt)
                 channelDistributor.spill(false);
 
-            Owned<IRowStream> distChannelStream = rhsDistributor->connect(queryRowInterfaces(rightITDL), right.getClear(), rightHash, NULL);
```

Comment:
I assume you meant line 2289 ?

I did leak before (but I didn't change it for that reason afair).

But it doesn't now afaict.
right used to be a Owned<IRowStream> and was set as : 
```
right.set(queryInputStream(1));
```

and then the connect line passed ownership via the getClear(), but connect() does not assume that and linked/release it.

Where as now right is not a linked object, ownership is not passed, connect() still link/releases it.